packages/material-ui/src/TableCell/TableCell.js

```diff
@@ -81,43 +82,58 @@ function TableCell(props, context) {
     ...other
   } = props;
 
-  const { table, tablelvl2 } = context;
-  let Component;
-  if (component) {
-    Component = component;
-  } else {
-    Component = tablelvl2 && tablelvl2.variant === 'head' ? 'th' : 'td';
-  }
+  return (
+    <TableContext.Consumer>
+      {table => {
+        return (
+          <Tablelvl2Context.Consumer>
```

Comment:
I don't follow the O-Notation with react components. Never heard that this is an issue.

Reply:
> I'm assuming it has an impact

That's the issue here. You are assuming alot about react internals without benchmarks. 

Just because it looks constant in the code does not mean that it has a constant performance impact. Especially since this port does not introduce new contexts. I did not introduce the consumption of 2 contexts instead of 1. That was already the case. Even worse: One context object will cause alot more rerenders than 2 context strings. That's what I hinted at in https://github.com/mui-org/material-ui/pull/13529#discussion_r231189907

I've gone through the context docs and react issues concerning context quite a bit these past few days and I couldn't find a section about performance impact of N contexts vs 1 merged one.

Reply:
It's fair, I will do a quick benchmark, see what % of differnence we have. Maybe it's négligeable.